### ISSUE TITLE
Explicitly set files not copied because they are marked as a reference

### ISSUE THREAD

**Issue by msfcolombo** [nuproj/nuproj#283]
I'm trying to package a helper tool in addition to my DLL. There is also a dependency to another package in the same solution. So my nuproj looks like this:

```
    <ProjectReference Include="$(SourceDir)\MyLibraryCore.nuproj" />

    <Library Include="output\lib\MyLibrary.dll" />

    <File Include="output\tool\MyTool.exe">
      <TargetPath>bin</TargetPath>
    </File>
    <File Include="output\tool\*dll">
      <TargetPath>bin</TargetPath>
    </File>

```
The problem is that my tool also depends on files that are on `MyLibraryCore`. They get removed by the `GetPackageFiles` target from nuproj, and as a result my tool misses a DLL when users try to execute it from the bin directory.

I think `<File>` elements should not be subject to have files skipped because they are present in dependencies. After all, `<File>` is away to obtain maximum control, according to documentation.

**Comment by AArnott** [comment]
Do you actually want to maintain the package reference to MyLibraryCore? I think it's a little unusual to have both a package reference and carry the payload locally, but if your package doubles as both a tool and a library then I guess it makes sense.

I sympathize with your desire for an ItemType that doesn't get filtered. I think `PackageFile` might work for you (just based on memory). Otherwise, I've been known to create `File` items in an MSBuild Target that runs *after* the filter is applied.

**Comment by msfcolombo** [comment]
The reason to reference MyLibraryCore is that nuproj will automatically add a nuget dependency in the Nupkg. Without the project reference, I'm forced to add an explicit `Dependency` element, and this causes me other troubles (see [here](http://stackoverflow.com/questions/42463551/why-nuproj-is-not-resolving-variable-inside-dependencyversion)).

**Comment by AArnott** [comment]
Yes, I understand the effect of a ProjectReference to a NuProj. My question is why you want both a NuGet dependency expressed as well as embed the content of that package. Usually you just do one or the other.

**Comment by msfcolombo** [comment]
Please look at issue text. The need is simple: When the user starts `MyTool.exe` from the `bin` directory, it should just work. This requires the `MyLibrary.dll` to be copied to `bin` directory, otherwise it will report a missing package. That's all!

BTW, `PackageFile` doesn't work. nuproj [nukes it](https://github.com/nuproj/nuproj/blob/master/src/NuProj.Targets/NuProj.targets#L629) just before removing dependencies.

**Comment by AArnott** [comment]
Yes, I read the text. My question is why you need *both* a package dependency *and* the embedded file. I understand why you need the embedded file. Are you sure you also need a package dependency? I'd be curious what your reason is. I know a reason that might apply, but I want to confirm first.

**Comment by msfcolombo** [comment]
I have two packages: the "core" which contains interfaces intended for unit testing, and the "library itself" which contains implementation and a helper tool. The "library itself" depends on the "core". I want Nuget to automatically fetch the "core" when someone installs the "library itself".

**Comment by AArnott** [comment]
OK, great. That sounds like a reasonable use case.
Can you take a look at [this sample](https://github.com/Microsoft/vs-threading/blob/v15.0/src/Microsoft.VisualStudio.Threading.NuGet/Microsoft.VisualStudio.Threading.NuGet.nuproj#L49) of how I force-add files to my package to see if it helps guide you how to workaround this?
